react/src/components/search-app/SearchBarComponent.jsx

```diff
@@ -0,0 +1,126 @@
+import Autocomplete from 'react-autocomplete';
+import React from 'react';
+
+
+export default class SearchBarComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: props.query,
+      suggestions: [],
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ query: nextProps.query });
+  }
+
+
+  static getItemValue(item) {
+    return item._index === 'work' ? item._source.title : item._source.name;
+  }
+
+
+  handleChange(event, query) {
+    this.setState({ query });
+    if (query.length >= 2) {
+      const http = new XMLHttpRequest();
+      const params = JSON.stringify({ query });
+      http.open('POST', '/api/v1/search', true);
```

Comment:
Regarding fetch: yes, I know ;)

Regarding exporting the URL: Yes, that's a good idea.